internal/addrs/module.go

```diff
@@ -177,6 +177,25 @@ func (m Module) configRemovableSigil() {
 	// Empty function so Module will fulfill the requirements of the removable interface
 }
 
+// ParseModule parses a module address from the given traversal,
+// which has to contain only the module address with no resource/data/variable/etc.
+// This function only supports module addresses without instance keys (as the
+// returned Module struct doesn't support instance keys) and will return an
+// error if it encounters one.
+func ParseModule(traversal hcl.Traversal) (Module, tfdiags.Diagnostics) {
+	mod, remain, diags := parseModulePrefix(traversal)
+	if len(remain) != 0 {
```

Comment:
Perhaps also check that `!diags.HasErrors()` ?

When diags.HasErrors(), any results from a function should be considered undefined.

Reply:
I see you did that for `ParseConfigResource` below

Reply:
Yep, missed this one, thank you! I was still getting used to diags instead of errors by the time I was writing this line :)

Fixed in ff1754d396ecae51c1ac70046647dc7b880e2e13